src/components/subscriptions/SubscriptionExpirationModal.jsx

```diff
@@ -0,0 +1,148 @@
+import React, { useContext } from 'react';
+import PropTypes from 'prop-types';
+import Cookies from 'universal-cookie';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { Modal, MailtoLink } from '@edx/paragon';
+
+import { SubscriptionContext } from './SubscriptionData';
+import {
+  SUBSCRIPTION_DAYS_REMAINING_MODERATE,
+  SUBSCRIPTION_DAYS_REMAINING_SEVERE,
+  SUBSCRIPTION_DAYS_REMAINING_EXCEPTIONAL,
+  SEEN_SUBSCRIPTION_EXPIRATION_MODAL_COOKIE_PREFIX,
+} from './constants';
+
+import Img from '../../components/Img';
+import edxLogo from '../../images/edx-logo.png';
+import { formatTimestamp } from '../../utils';
+
+export const MODAL_DIALOG_CLASS_NAME = 'subscription-expiration';
+
+
+export function BaseSubscriptionExpirationModal({ enterpriseSlug, enableCodeManagementScreen }) {
+  const { details } = useContext(SubscriptionContext);
+  const { daysUntilExpiration, expirationDate } = details;
+
+  const renderTitle = () => (
+    <small>Renew your subscription</small>
+  );
+
+  const renderBody = () => (
+    <React.Fragment>
+      <p>
+        Your current subscription is set to expire in {daysUntilExpiration} days.
+        In order to minimize course access disruptions for your learners, make sure your invoice has
+        been completed.
+      </p>
+      <p>
+        If you have questions or need help, please contact the edX Customer Success team at
+        {' '}
+        <MailtoLink to="customersuccess@edx.org">customersuccess@edx.org</MailtoLink>.
+      </p>
+      <i>
+        Access expires on {formatTimestamp({ timestamp: expirationDate })}
+      </i>
+    </React.Fragment>
+  );
+
+  const renderExpiredBody = () => (
+    <React.Fragment>
+      <Img className="w-25 my-5 mx-auto d-block" src={edxLogo} alt="edX logo" />
+      <p>
+        Your subscription license expired on {formatTimestamp({ timestamp: expirationDate })}.
+        To access your subscription management page contact edX and reactivate your subscriptions.
+      </p>
+      <p>
+        What to do next?
+      </p>
+      <ul>
+        <li>
+          To reactivate your subscriptions please contact the edX Customer Success team at
```

Comment:
Nit: "subscription" singular, or "subscription(s)", if you're afraid of commitment.

Reply:
I am indeed afraid